.github/workflows/docker-devito.yml

```diff
@@ -0,0 +1,102 @@
+name: Publish devito docker images from compiler bases
+
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - master         # Push events on master branch
+
+jobs:
+  deploy-devito:
+    runs-on: ${{ matrix.runner }}
+    env:
+      DOCKER_BUILDKIT: "1"
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - base: "bases:nvidia-nvc"
+            tag: 'nvidia-nvc'
+            flag: '--gpus all'
+            test: 'tests/test_gpu_openacc.py tests/test_gpu_common.py'
+            runner: ["self-hosted", "gpu", "docker"]
+
+          - base: "bases:nvidia-clang"
+            tag: 'nvidia-clang'
+            flag: '--gpus all'
+            test: 'tests/test_gpu_openmp.py tests/test_gpu_common.py'
+            runner: ["self-hosted", "gpu", "docker"]
+
+          # Runtime gpu flags from https://hub.docker.com/r/rocm/tensorflow/
+          - base: "bases:amd"
+            tag: 'amd'
+            flag: '--network=host --device=/dev/kfd --device=/dev/dri --ipc=host --group-add video'
+            test: 'tests/test_gpu_openmp.py tests/test_gpu_common.py'
```

Comment:
drop test_gpu_common here or it will blow up memory?

yep, one more reason to perform a workflow-dispatch beforehand

Reply:
Well this one is tricky to test because it's triggered on merge or release